src/plugins/ai_assistant_management/observability/public/routes/components/knowledge_base_bulk_import_flyout.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCode,
+  EuiCodeBlock,
+  EuiFilePicker,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiHorizontalRule,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { KnowledgeBaseEntry } from '../../../common/types';
+import { useImportKnowledgeBaseEntries } from '../../hooks/use_import_knowledge_base_entries';
+
+export function KnowledgeBaseBulkImportFlyout({ onClose }: { onClose: () => void }) {
+  const { mutateAsync, isLoading } = useImportKnowledgeBaseEntries();
+
+  const filePickerId = useGeneratedHtmlId({ prefix: 'filePicker' });
+
+  const [files, setFiles] = useState<File[]>([]);
+
+  const onChange = (file: FileList | null) => {
+    setFiles(file && file.length > 0 ? Array.from(file) : []);
+  };
+
+  const handleSubmitNewEntryClick = async () => {
+    let entries: Array<Omit<KnowledgeBaseEntry, '@timestamp'>> = [];
+    const text = await files[0].text();
+
+    const elements = text.split('\n').filter(Boolean);
+
+    try {
+      entries = elements.map((el) => JSON.parse(el)) as Array<
+        Omit<KnowledgeBaseEntry, '@timestamp'>
+      >;
+    } catch (_) {
+      /* empty */
+    }
+
+    try {
+      await mutateAsync({ entries });
+      onClose();
+    } catch (_) {
+      /* empty */
```

Comment:
I'm guessing this failure is toasted by the react-query code? Perhaps the comment can hint at that?

Reply:
Sure